zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -93,6 +93,14 @@ public NotebookRepoSync(ZeppelinConfiguration conf) {
       LOG.info("No storage could be initialized, using default {} storage", defaultStorage);
       initializeDefaultStorage(conf);
     }
+    // sync for anonymous mode on start
+    if (getRepoCount() > 1 && conf.getBoolean(ConfVars.ZEPPELIN_ANONYMOUS_ALLOWED)) {
+      try {
+        sync(AuthenticationInfo.ANONYMOUS);
+      } catch (IOException e) {
+        LOG.error("Couldn't sync on start ", e);
```

Comment:
@felixcheung good point, actually this is done before the server actually starts so at this time user wouldn't have his window open either, meaning it would be difficult to notify the user on that one. However next step plan on this feature is to move it to frontend and have button in `NotebookRepo` menu. Then it will be more self-explanatory and possible to display status on failure.